.circleci/config.yml

```diff
@@ -0,0 +1,230 @@
+---
+
+version: 2.1
+
+.install_bundler: &install_bundler
+  run:
+    name: Install a specific bundler version
+    command: gem install bundler -v 1.17.1
+
+.copy_current_gemfile: &copy_current_gemfile
+  run:
+    name: Cat current Gemfile to a specific file so caching works
+    command: cat "$BUNDLE_GEMFILE.lock" > current_gemfile.lock
+
+.restore_cache: &restore_cache
+  restore_cache:
+    keys:
+      - bundler-dependencies-{{ checksum "current_gemfile.lock" }}-{{ .Environment.CIRCLE_JOB }}
+
+.install_dependencies: &install_dependencies
+  run:
+    name: Install dependencies
+    command: bundle install --without development --path /home/circleci/project/vendor/bundle --retry 3 --jobs 3
+
+.save_cache: &save_cache
+  save_cache:
+    key: bundler-dependencies-{{ checksum "current_gemfile.lock" }}-{{ .Environment.CIRCLE_JOB }}
+    paths:
+      - vendor/bundle
+
+.run_tests: &run_tests
+  run:
+    name: Run tests
+    command: bundle exec rake test TESTOPTS="--verbose"
+
+.test_steps: &test_steps
+  - checkout
+  - *install_bundler
+  - *copy_current_gemfile
+  - *restore_cache
+  - *install_dependencies
+  - *save_cache
+  - *run_tests
+
+jobs:
+  ruby23rails50:
+    docker:
+      - image: circleci/ruby:2.3.7-node-browsers
```

Comment:
I don't think we need the `-browser` variants here because there are no browser specs in this gem, and I'm not fully sure we need the `-node` variants either. Can you check?

Reply:
Sure! I'll give it a try without both. I couldn't confirm if one exists without `-browsers` and `-node` but I think it does.

Reply:
It worked! That's awesome, thanks! Fastest build yet. I'll squash that change into the first commit.